Documentation for the changes in private registry credentials (#509)
okteto/docs: src/content/administration/dashboard.mdx

## Registry Credentials

> This section of the Admin dashboard allows you to manage credentials for private registries that you want to use in your Okteto instance. Those credentials are injected by Okteto in any operation (build, deploy, etc...) done via CLI or the UI, so developers don't need access to them. You only need to configure your credentials here, and all the developers will be able to interact with those registries while developing with Okteto. If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes, so the cluster can pull images from those registries.

Reviewer: ```suggestion
This section of the Admin dashboard lets you manage credentials for private registries to use in your Okteto instance. The credentials you specify here are injected by Okteto into any operation (e.g. build, deploy) you perform (UI or CLI) so developers don't need access to them. You only need to configure your credentials here and all developers in your Okteto instance will be able to interact with those registries while developing with Okteto. If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes so the cluster can pull images from those registries.
```

Some minor suggested changes 🙂

For each configured registry, you will see a row with the type (Static or AWS) and the registry hostname. The 2 types of credentials supported are:

> - Static: To configure registries with static username and password like Docker Hub.

Reviewer: ```suggestion
- Static: Registry configuration using static username and password credentials (e.g. Docker Hub)
```

> This section of the Admin dashboard lets you manage credentials for private registries to use in your Okteto instance. The credentials you specify here are injected by Okteto into any operation (e.g. build, deploy) you perform (UI or CLI) so developers don't need access to them. You only need to configure your credentials here and all developers in your Okteto instance will be able to interact with those registries while developing with Okteto. If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes so the cluster can pull images from those registries.

Reviewer: ```suggestion
In this section of the Admin dashboard you can manage the private registry credentials for your Okteto instance. The credentials you configure here are injected by Okteto into any operation (e.g. build, deploy) performed by developers so they don't need direct access to the credentials. Once you configure the credentials here all developers in your Okteto instance will be able to use the respective registries while developing with Okteto. 

If the setting [`daemonset.configurePrivateRegistriesInNodes.enabled`](/self-hosted/administration/configuration.mdx/#daemonset) is enabled, Okteto's daemonset will also configure these credentials in the nodes so the cluster can also pull images from those registries.
```

I took a stab at breaking this up a bit to make it a bit easier to digest and also make it a bit more concise

> - AWS: To configure [ECR registries](https://docs.aws.amazon.com/ecr/index.html) using Access Key and Access Secret. In this case, Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured. The hostname of the registry has to end with `.amazonaws.com`.

Reviewer: ```suggestion
- AWS: [ECR registry](https://docs.aws.amazon.com/ecr/index.html) configuration using Access Key and Access Secret. Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured. The hostname of the registry must end with `.amazonaws.com`.
```

<p align="center"><Image src="/docs/administration/images/registry_credentials_view.png" alt="Registry credentials view" width="900" /></p>

> Apart of adding new credentials, this view also allows you to rotate those credentials or delete them if you don't need them anymore. For sensitive data you only see the last few characters as a hint to verify the value.

Reviewer: ```suggestion
This admin view also enables you to rotate your registry credentials or delete them if you don't need them anymore. For sensitive data you will only see the last few characters as a hint to verify the value.
```

<p align="center"><Image src="/docs/administration/images/registry_credentials_detail.png" alt="Registry credentials detail" width="700" /></p>

> > Be aware that any change you do in those credentials might take a few minutes to be propagated to all the components in the cluster.

Reviewer: ```suggestion
> Be aware that any changes you make to credentials may take a few minutes to propagate to all of the components in the cluster.
```

For each configured configured registry, you will see a row with the type (Static or AWS) and the registry hostname. The 2 types of credentials supported are:
- Static: To configure registries with static username and password like Docker Hub.
- AWS: To configure [ECR registries](https://docs.aws.amazon.com/ecr/index.html) using Access Key and Access Secret. In this case, Okteto takes care of refreshing docker credentials every 4 hours using the key and secret configured. The hostname of the registry has to end with `.amazonaws.com`.

<p align="center"><Image src="/docs/administration/images/registry_credentials_view.png" alt="Registry credentials view" width="900" /></p>

Apart of adding new credentials, this view also allows you to rotate those credentials or delete them if you don't needed them anymore. For sensitive data you only see some of the last characters as a hint to verify the value.

<p align="center"><Image src="/docs/administration/images/registry_credentials_detail.png" alt="Registry credentials detail" width="700" /></p>

> Be aware that any change you do in those credentials might take a few minutes to be propagated to all the components in the cluster.

> ### Advanced configuration

Reviewer 2: I would move this section to another doc. Maybe a doc in `Self-hosted > Configuration`? Or in the community forum?

Reviewer 2: Maybe a section about CRDs in general. We should document the git catalog CRD in case someone prefers to define the catalog in code.
But I feel CRDs is too low level to be included in a doc that explains the Admin View

Author: I added it here because there was not specific section for private registries or CRDs. I feel that one specific for private registry might be a small without too many content. One about CRDs might make sense as we can talk about others

Author: @pchico83  Moved to its own page. Take a look when you have a chance and let me know what you think

    userSecretRef:
      key: <field in referenced secret> # key of the secret which contains the username
      name: <secret-name> # name of the secret where the username is stored
```

In the case of AWS, where you have to specify the Access Key and Access Secret, the process is similar to the static credentials one. You will need to create a secret with those values, and then, create a CRD like this (the name of the resource has to be the hostname of the registry to be configured and it has to end with `.amazonaws.com`):

```yaml
apiVersion: admin.okteto.com/v1
kind: Registry
metadata:
  labels:
    app.kubernetes.io/managed-by: okteto
  name: 111122223333.dkr.ecr.eu-central-1.amazonaws.com # hostname of the registry to be configured
  namespace: okteto
spec:
  awsIamUser:
    accessKeyIDSecretRef:
      key: <field in referenced secret> # key of the secret which contains the access key
      name: <secret-name> # name of the secret where the access key is stored
    secretAccessKeySecretRef:
      key: <field in referenced secret> # key of the secret which contains the access secret key
      name: <secret-name> # name of the secret where the access secret is stored
```

> Last thing worth to mention is related to the label `app.kubernetes.io/managed-by: okteto` in the CRDs. Adding that label means that the credentials will be editable from the UI, so any admin could come to this view and edit them. If the lifecycle of those credentials is managed directly by you outside Okteto, you can remove that label. Okteto will keep considering those credentials to be used, and they will keep showing in this view but as read only.

Reviewer 2: I think this is always the case when using CRDs. I would remove the label `app.kubernetes.io/managed-by: okteto` from the samples and its explanation

Reviewer 3: +1. I don't think we should encourage the use of this label to resources created directly via `kubectl`. This could bring problems. With this label enabled (since it's managed by us) we assume a single secret with "username" and "password" keys and a predefined secret name. If they use this label and use different keys or a different secret name it will work initially but it will be editable through the UI and once edited through the UI it will create a new secret.

To recap, if they want to add the label they need to:

1. Use the same secret for user/password
2. The secret keys **must** be `username` and `password`
3. The secret name must be: `{host}-regcreds`

Author: Ok, agree to explain it in in the other way, but I would refer somehow to that label or explain it. If someone wants to migrate their current registries defined here and migrated automatically by Okteto to their git ops approach, they would have to remove that label or it will have problems if change the value directly from the UI

Author: @maroshii Related to this:

```
To recap, if they want to add the label they need to:

* The secret keys must be username and password
* The secret name must be: {host}-regcreds
```

Aren't we reusing the secret and field names defined in the CRD when updating or managing the credentials?

Reviewer 3: Yes we are and the read path should not break but when we create/update we make some assumptions. For eg, we assume they are stored in a single secret and we only read `reg.Spec.Static.UserSecretRef.Name`.

The whole point of that label was to be able to make those assumptions and make change without affecting users. I think we should keep that property and not expose these internals.

Author: Ok, thanks for the explanation. I moved all this to another section. It's own page and removed the reference for the label. Take a look when you have a chance
